src/interpreter/plugin/AddressPlugin.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * @license
+ * Copyright (c) 2023 Handsoncode. All rights reserved.
+ */
+
+import {columnIndexToLabel} from '../../parser/addressRepresentationConverters'
+import {ProcedureAst} from '../../parser'
+import {InterpreterState} from '../InterpreterState'
+import {InterpreterValue} from '../InterpreterValue'
+import {FunctionArgumentType, FunctionPlugin, FunctionPluginTypecheck, ImplementedFunctions} from './FunctionPlugin'
+import {CellError, ErrorType} from '../../Cell'
+import {ErrorMessage} from '../../error-message'
+import {Maybe} from '../../Maybe'
+
+enum AbsStyle {
+  FullyAbsolute = 1,
+  RowAbsoluteColRelative = 2,
+  RowRelativeColAbsolute = 3,
+  FullyRelative = 4,
+}
+
+export class AddressPlugin extends FunctionPlugin implements FunctionPluginTypecheck<AddressPlugin> {
+  public static implementedFunctions: ImplementedFunctions = {
+    'ADDRESS': {
+      method: 'address',
+      parameters: [
+        {argumentType: FunctionArgumentType.NUMBER},
+        {argumentType: FunctionArgumentType.NUMBER},
+        {argumentType: FunctionArgumentType.NUMBER, optionalArg: true, defaultValue: 1, minValue: 1, maxValue: 4},
+        {argumentType: FunctionArgumentType.BOOLEAN, optionalArg: true, defaultValue: true},
+        {argumentType: FunctionArgumentType.STRING, optionalArg: true},
+      ]
+    },
+  }
+
+  private verifyAddressArguments(row: number, col: number, abs: number, useA1Style: boolean): Maybe<CellError> {
+    if (useA1Style) {
+      if ((row < 1) || (col < 1)) {
+        return new CellError(ErrorType.VALUE, ErrorMessage.LessThanOne)
+      }
+    } else {
+      if (AbsStyle.FullyAbsolute == abs) {
+        if ((row < 1) || (col < 1)) {
+          return new CellError(ErrorType.VALUE, ErrorMessage.LessThanOne)
+        }
+      } else if (AbsStyle.RowAbsoluteColRelative == abs) {
+        if (row < 1) {
+          return new CellError(ErrorType.VALUE, ErrorMessage.LessThanOne)
+        }
+      } else if (AbsStyle.RowRelativeColAbsolute == abs) {
+        if (col < 1) {
+          return new CellError(ErrorType.VALUE, ErrorMessage.LessThanOne)
+        }
+      } else if (AbsStyle.FullyRelative == abs) {
+        // no range checks needed for AbsStyle.FullyRelative
+      }
+    }
+    return undefined

```

Comment:
```suggestion
```

Reply:
I prefer the explicit return statement as this function is not `void`